http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -186,9 +189,11 @@ class LeoPubsubMessageSubscriber[F[_]](
       }
     } yield ()
 
-    res.handleErrorWith(e =>
-      logger.error(e)("Fail to process pubsub message for some reason") >> F.delay(event.consumer.ack())
-    )
+    res
+      .handleErrorWith(e =>
+        logger.error(e)("Fail to process pubsub message for some reason") >> F.delay(event.consumer.ack())
+      )
+      .guarantee(F.delay(span.end()))
```

Comment:
`guarantee` comes from cats-effect [MonadCancel](https://typelevel.org/cats-effect/docs/typeclasses/monadcancel), it works like `finally` :)